File: libs/language/src/lib/mapping/__tests__/schema/personDictionaryExample.ts
import z from 'zod';
import { withTypeBlueId } from '../../../../schema/annotations';
import { personSchema } from './person';

export const personDictionaryExampleSchema = withTypeBlueId(
  'PersonDictionary-BlueId'
)(
  z.object({
    team1: z.map(z.string(), personSchema),
    team2: z.map(z.string(), personSchema),
    team3: z.map(z.number(), personSchema),
  })
);

File: references/java-sdk/src/test/java/blue/language/mapping/model/PersonDictionaryExample.java
package blue.language.mapping.model;

import blue.language.model.*;

import java.util.Map;

@TypeBlueId("PersonDictionary-BlueId")
public class PersonDictionaryExample {
    public Map<String, ? extends Person> team1;
    public Map<String, Person> team2;
    public Map<Integer, Person> team3;
}